Assets/SEE/VCS/GitVersionControl.cs

```diff
@@ -0,0 +1,187 @@
+using LibGit2Sharp;
+using System.Linq;
+using System.IO;
+using System;
+using System.Collections.Generic;
+using UnityEngine.Assertions;
+using UnityEngine;
+
+namespace SEE.VCS
+{
+    /// <summary>
+    /// Implements the functionality of <see cref="IVersionControl"/> for Git.
+    /// </summary>
+    internal class GitVersionControl : IVersionControl
+    {
+        /// <summary>
+        /// Constructor setting up the repository access.
+        /// </summary>
+        /// <param name="repositoryPath">the path to a Git repository</param>
+        /// <exception cref="ArgumentNullException">throw in case <paramref name="repositoryPath"/>
+        /// is null or just whitespace</exception>
+        internal GitVersionControl(string repositoryPath)
+        {
+            if (string.IsNullOrWhiteSpace(repositoryPath))
+            {
+                throw new ArgumentNullException("Repository path must not be null or empty.");
+            }
+            this.repositoryPath = repositoryPath;
+            repo = new(repositoryPath);
+        }
+
+        /// <summary>
+        /// The repository.
+        /// </summary>
+        private readonly Repository repo;
+
+        /// <summary>
+        /// The path to the root of the repository.
+        /// </summary>
+        private readonly string repositoryPath;
+
+        /// <summary>
+        /// See <see cref="IVersionControl.Show(string, string)"/>.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">throw in case <paramref name="fileName"/>
+        /// or <paramref name="commitID"/> are null or just whitespace</exception>
+        public string Show(string fileName, string commitID)
+        {
+            if (string.IsNullOrWhiteSpace(fileName))
+            {
+                throw new ArgumentNullException("Path must not be null or empty.");
+            }
+            if (string.IsNullOrWhiteSpace(commitID))
+            {
+                throw new ArgumentNullException("Commit ID must not be null or empty.");
+            }
+            Blob blob = repo.Lookup<Blob>($"{commitID}:{Path.GetRelativePath(repositoryPath, fileName).Replace("\\", "/")}");
```

Comment:
In theory, paths in Unix-based systems can contain backslashes that are not path separators. It might be okay to ignore this case though and hope we never encounter this.

Reply:
Oh, well, let's hope this never happens. Some things should be strictly forbidden.